pkg/cinderapi/volumes.go

```diff
@@ -24,19 +26,46 @@ func GetVolumes(parentName string, name string, extraVol []cinderv1beta1.CinderE
 		},
 	}
 
-	return append(cinder.GetVolumes(parentName, false, extraVol, cinder.CinderAPIPropagation), backupVolumes...)
+	for _, secretName := range secretNames {
+		secretVol := corev1.Volume{
+			Name: secretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  secretName,
+					DefaultMode: &config0640AccessMode,
+				},
+			},
+		}
+
+		volumes = append(volumes, secretVol)
+	}
+
+	return append(cinder.GetVolumes(parentName, false, extraVol, cinder.CinderAPIPropagation), volumes...)
 }
 
 // GetInitVolumeMounts - Cinder API init task VolumeMounts
-func GetInitVolumeMounts(extraVol []cinderv1beta1.CinderExtraVolMounts) []corev1.VolumeMount {
+func GetInitVolumeMounts(secretNames []string, extraVol []cinderv1beta1.CinderExtraVolMounts) []corev1.VolumeMount {
+
+	initVolumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "config-data-custom",
+			MountPath: "/var/lib/config-data/custom",
+			ReadOnly:  true,
+		},
+	}
+
+	for idx, secretName := range secretNames {
+		secretMount := corev1.VolumeMount{
+			Name: secretName,
+			// Each secret needs its own MountPath
+			MountPath: "/var/lib/config-data/secret-" + strconv.Itoa(idx),
+			ReadOnly:  true,
+		}
```

Comment:
This also seems common to multiple files

Reply:
Deduped in latest patchset